### PR TITLE
feat: add swc package to monorepo group

### PR DESCRIPTION
**Pull request by nissy-dev**
## Changes

Adds a new group preset that groups all swc related packages together.

## Context

SWC is one of the most popular compiler for JS/TS. It is developed at https://github.com/swc-project/swc, and also the plugin is developed at https://github.com/swc-project/plugins. SWC plugin highly depends on SWC core package, so we need to update SWC core and SWC plugin at the same time in many cases. I thought this group might be helpful for others as well.

Example plugins: @swc/plugin-styled-components, @swc/plugin-emotion, @swc/plugin-relay....

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
it should be grouped by source url, checkout monorepo.ts

**Comment by nissy-dev** [comment]
Thank you for reviews! I refactored using grouping by source url.

File: lib/config/presets/internal/monorepo.ts
Comment (HonkingGoose) [inline lib/config/presets/internal/monorepo.ts:230]: Other presets in this file have one URL in them. The current preset has four URLs. I'll let the maintainers decide which option is best.

### Option 1: group whole `swc-project` repository

```ts
'swc-project': 'https://github.com/swc-project',
```

### Option 2: separate entries for each repository

```ts
'swc-project/swc': 'https://github.com/swc-project/swc',
'swc-project/plugins': 'https://github.com/swc-project/plugins',
'swc-project/jest': 'https://github.com/swc-project/jest',
'swc-project/cli': 'https://github.com/swc-project/cli',
```

### Option 3: do nothing

Maybe this is a valid config, and I'm confusing myself somehow. In that case you can ignore me. 😄

Comment (viceice) [reply lib/config/presets/internal/monorepo.ts:230]: do they have more repos which should not grouped? then we stay as is, otherwise we can group by organization url

Comment (nissy-dev) [reply lib/config/presets/internal/monorepo.ts]: Grouping by oraganization url seem to be a fine for me, so I refactored in [8760e3c](https://github.com/renovatebot/renovate/pull/21431/commits/8760e3c5251916a9426d8bf3c08930c99eec1b68).

File: lib/config/presets/internal/monorepo.ts
Comment (viceice) [inline lib/config/presets/internal/monorepo.ts:266]:
```suggestion
  swc: 'https://github.com/swc-project/',
```
so `swc-project1` isn't matched 😏

**Review by viceice — APPROVED** [review]

**Pull request merged**
